activestorage/lib/active_storage/service/azure_storage_service.rb

```diff
@@ -197,5 +203,16 @@ def handle_errors
       def custom_metadata_headers(metadata)
         metadata.transform_keys { |key| "x-ms-meta-#{key}" }
       end
+
+      def custom_checksum_headers(checksum)
+        case checksum_algorithm
+        when :MD5
+          { "Content-MD5" => checksum }
+        when :CRC64
+          { "x-ms-content-crc64" => checksum }
```

Comment:
I'm not very familiar with Active Storage. This `checksum` is only used for upload? Not stored in DB or anything? 

In otherwords, can we switch to other algorithms, or does it require a complex data migration for users?

Reply:
It is 
- used by the cloud service to validate the uploaded data or return an error.
- saved locally as part of `active_storage_blobs` table in `checksum` column. I can add another column to store the type to support migrations/transitioning from one type to another.

Reply:
> saved locally as part of active_storage_blobs table in checksum column.

Ah that's what I feared.

Rather than another column we could do a prefix, e.g. `md5:caffe42343...`, and if no prefix we can assume it's MD5 which would allow for a smooth transition hopefully? (again I apologize here, I never used Active Storage beyond basic feature so I'm not as familiar with its internal than with other Rails parts).

Reply:
There have been a few migrations for AS in the past (https://github.com/rails/rails/tree/main/activestorage/db/update_migrate). This one in particular is very close what would be needed here https://github.com/rails/rails/blob/fa9cf269191c5077de1abdd1e3f934fbeaf2a5d0/activestorage/db/update_migrate/20190112182829_add_service_name_to_active_storage_blobs.rb

It seems fine to add a new column for this? I guess the question would be if that column is public. In my application, I grab that checksum and decode it to display the actual md5 hash in the UI which would break when it contains additional data. Mind you, it would also break when it contains a different hash type, but less so and it would be more egonomic to handle if there were two dedicated columns for this.

Reply:
I also played around with a branch that added a new column for the algorithm attribute, but I prefer this approach with serialize due to backwards compatibility being more simple.